File: sdk/python/openbrain_sdk/examples/mcp_e2e.py
import os
import time

from openbrain_sdk import OpenBrainMcpClient


def main() -> None:
    scope = "sdk-usage-proof"
    obj_id = f"artifact-{int(time.time())}"

    env = dict(os.environ)
    env["OPENBRAIN_EMBED_PROVIDER"] = "fake"

    client = OpenBrainMcpClient.spawn_openbrain_mcp(openbrain_path="openbrain", env=env)
    try:
        ping = client.call_tool("openbrain.ping", {})
        print(f"version: {ping.get('version')}")

        write = client.call_tool(
            "openbrain.write",
            {
                "objects": [
                    {
                        "type": "artifact",
                        "id": obj_id,
                        "scope": scope,
                        "status": "canonical",
                        "spec_version": "0.1",
                        "tags": ["sdk", "usage-proof"],
                        "data": {"summary": "MCP usage proof"},
                        "provenance": {"actor": "sdk-usage-proof", "method": "mcp-e2e"},
                    }
                ]
            },
        )

        results = write.get("results", [])
        ref = results[0]["ref"] if results else obj_id
        read = client.call_tool("openbrain.read", {"scope": scope, "refs": [ref]})
        print(f"read objects: {len(read.get('objects', []))}")

        embed = client.call_tool(
            "openbrain.embed.generate",
            {"scope": scope, "target": {"ref": ref}, "model": "default"},
        )
        print(f"embedding reused: {embed.get('reused')}")

        semantic = client.call_tool(
            "openbrain.search.semantic",
            {
                "scope": scope,
                "query": "usage proof",
                "top_k": 5,
                "embedding_provider": "fake",
                "embedding_model": "default",
                "embedding_kind": "semantic",
            },
        )
        print(f"semantic matches: {len(semantic.get('matches', []))}")

        try:
            pack = client.call_tool(
                "openbrain.memory.pack",
                {"scope": scope, "task_hint": "Summarize current decisions"},
            )
            print(f"pack summary: {pack['pack']['summary']}")
        except Exception as exc:
            print(f"memory.pack skipped: {exc}")
    finally:
        client.close()


if __name__ == "__main__":
    main()

File: sdk/typescript/openbrain-sdk/examples/mcp_e2e.ts
import { OpenBrainMcpClient } from "../src/mcp.js";

async function main() {
  const scope = "sdk-usage-proof";
  const id = `artifact-${Date.now()}`;

  const client = await OpenBrainMcpClient.spawnOpenBrainMcp({
    openbrainPath: "openbrain",
    env: { ...process.env, OPENBRAIN_EMBED_PROVIDER: "fake" },
  });

  try {
    const ping = await client.callTool("openbrain.ping", {});
    console.log("version:", ping.version);

    const write = await client.callTool("openbrain.write", {
      objects: [
        {
          type: "artifact",
          id,
          scope,
          status: "canonical",
          spec_version: "0.1",
          tags: ["sdk", "usage-proof"],
          data: { summary: "MCP usage proof" },
          provenance: { actor: "sdk-usage-proof", method: "mcp-e2e" },
        },
      ],
    });

    const ref = write.results[0]?.ref ?? id;
    const read = await client.callTool("openbrain.read", { scope, refs: [ref] });
    console.log("read objects:", read.objects.length);

    const embed = await client.callTool("openbrain.embed.generate", {
      scope,
      target: { ref },
      model: "default",
    });
    console.log("embedding reused:", embed.reused);

    const semantic = await client.callTool("openbrain.search.semantic", {
      scope,
      query: "usage proof",
      top_k: 5,
      embedding_provider: "fake",
      embedding_model: "default",
      embedding_kind: "semantic",
    });
    console.log("semantic matches:", semantic.matches.length);

    try {
      const pack = await client.callTool("openbrain.memory.pack", {
        scope,
        task_hint: "Summarize current decisions",
      });
      console.log("pack summary:", pack.pack.summary);
    } catch (error) {
      const message = error instanceof Error ? error.message : String(error);
      console.warn("memory.pack skipped:", message);
    }
  } finally {
    await client.close();
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
